### ISSUE TITLE
`unnecessary_null_checks` conflicts with the new `NULLABLE_ARGUMENT_TO_NON_NULL_TYPE`

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#4077]
There is a new hint proposed for https://github.com/dart-lang/sdk/issues/51367.

```dart
void f(int? i) => Future<int>.value(i);
```

The potential fix of making the argument not-nullable (by null-check `!`) would trigger `unnecessary_null_checks`.

This issue is a request that the lint ignores the argument of `Future.value` and `Completer.complete`.

This was detected during trying to fix Flutter for the hint.

**Comment by srawlins** [comment]
I agree these two cases should be excepted. Making the null-check explicit at the call site is better than keeping it implicit.
